### PR TITLE
Added option to display icons instead of text in context actions for iOS

### PR DESCRIPTION
**Pull request by Sebastian1989101**
### Description of Change ###

/!\ This is the same PR as https://github.com/xamarin/Xamarin.Forms/pull/6382 just implemented in a more dynamic way but in this case the More ContextAction could not changed by the app developer (because it is never an icon) and it requires more changes to Xamarin.Forms (MenuItem additional interface) - I would prefer https://github.com/xamarin/Xamarin.Forms/pull/6382 as 2nd thoughts  /!\

I've added a enum to choose if ContextActions within a ListView should be displayed as Icon or Text. This can be activated for each individual MenuItem. 

### Issues Resolved ### 

None

### API Changes ###

Added:
 - IElementConfiguration<MenuItem> to MenuItem
- Added On<iOS>().SetContextActionDisplay(ContextActionDisplay.Icon) for MenuItem

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None with default settings. When ContextActionDisplay.Icon is set for MenuItem they will appear as Icon instead of text in iOS. 

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Added "Context Action Display Test" to Control Gallery which includes one Icon ContextAction and one Text based. 

### PR Checklist ###

- [ ] Has automated tests 
- [x] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### Additional noted ###

Maybe this option should be moved to the ViewCell instead. This would prevent a base class to change (in this case MenuItem which now has an additional interface) and the More-Action could also change to an Icon (maybe the UITabBarSystemItem.More?) but it will remove some freedom which should be displayed as Icon and which one should not (at least if there is any reason for this to happen?). 

/!\ This is the same PR as https://github.com/xamarin/Xamarin.Forms/pull/6382 just implemented in a more dynamic way but in this case the More ContextAction could not changed by the app developer (because it is never an icon) and it requires more changes to Xamarin.Forms (MenuItem additional interface) - I would prefer https://github.com/xamarin/Xamarin.Forms/pull/6382 as 2nd thoughts  /!\

### CODE REVIEW

**Comment by Sebastian1989101** [comment]
Updated initial PR text

File: Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/MenuItem.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
+﻿namespace Xamarin.Forms.PlatformConfiguration.iOSSpecific
+{
+	using FormsElement = Forms.MenuItem;
+
+	public static class MenuItem
+	{
+		public static readonly BindableProperty ContextActionDisplayProperty = BindableProperty.Create(nameof(ContextActionDisplay), typeof(ContextActionDisplay), typeof(MenuItem), ContextActionDisplay.Text);
```
Comment (hartez) [inline Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/MenuItem.cs]: `typeof(MenuItem)` - this is likely referring to this static class; this should probably be `typeof(Forms.MenuItem)` or `typeof(FormsElement)`.

**Comment by geskill** [comment]
Should add the TextAndIcon or simply „Both“ value directly since adding the iOS implementation is done in a sec

**Comment by PureWeen** [comment]
if it is indeed easy and @Sebastian1989101 wants to try it 👍 

But it's not something I would hold up this PR for.

**Comment by rmarinho** [comment]
Hey @Sebastian1989101 do you want to get back to apply the changes requested or should we take over?

**Comment by Sebastian1989101** [comment]
@rmarinho Feel free to take it if you want. I don't get the time currently to make the requested modifications.

**Comment by rmarinho** [comment]
Since we have SwipeView now we will work on that feature there.

**Pull request closed without merging**